### ISSUE TITLE
ComfyUI Not using GPU

### ISSUE THREAD

**Issue by UndergroundBeauty** [comfyanonymous/ComfyUI#3970]
### Your question

MY ComfyUI seems to be using CPU rather than my GPU. I have CUDA Version 12.4 installed (verified by nvidia-smi in power shell). What troubleshooting methods can I take to ensure comfyui uses my GPU?

**Comment by bluevisor** [comment]
try 12.1, and you'll also need corresponding torch and xformers, etc...

**Comment by ltdrdata** [comment]
It seems that you are executing `run_cpu.bat` instaed of `run_nvidia.bat`

**Comment by UndergroundBeauty** [comment]
Apologies, now I am realizing it is specific to workflows. Some workflows use GPU, some (my video faceswap workflow using reactor) is using system ram and crashing my computer. Thoughts? Is this a node specific issue?

**Comment by mcmonkey4eva** [comment]
Yes, that'd be an issue specific to the custom nodes that are running on CPU

**Comment by Statvar** [comment]
I know this threads old but I was having this issue as well and still don't know what initially caused it. I went to my python_embeded folder and launched cmd through there and uninstalled pytorch and installed the latest version. I did this bc I noticed when I checked my torch version it said mine was 2.5.0+cpu. I used the install command here https://pytorch.org/get-started/locally/ after uninstalling my current version. To check your version of torch type, 'pip show torch' My version now says 2.7.0.dev20250118+cu124. Some people say to run the update_comfyui_and_python_dependencies to fix it but that didn't work for me so I had to resort to doing in manually. Hope that helps somebody else.
